Repair Kits [READY]

Dark-Umbrella opened the pull request:
## About The Pull Request

The PR adds 4 new items - repair kits.
2 of them are homemade, low-quality versions of the other 2, which are created exclusively by craftsmen. Kits fix things as a percentage, proportionally taking self-damage during the process.
Repair kits do not require any skills to use.

<img width="900" height="458" alt="image" src="https://github.com/user-attachments/assets/9bf2cf24-ef26-4083-9d8f-c77a0c3ae105" />

Sewing kit - created by a sewing expert using 3 strips of fabric and 2 pieces of treated leather. It can repair 7 (700%) items from a completely broken state. Why do it when you have a needle? The kit allows you to fix things directly on yourself, without having to find a table.

Armor plates - created by a blacksmithing expert, from 2 ingots of steel, 1 ingot of iron and 1 treated leather. Can fix 10 things (1000%). 2 pieces are created at once to compensate for the high cost of the recipe. Requires a table for repair.

Fabric Patch - created by Journeyman crafting, using 2 strips of fabric, 2 fibers and 1 rope. The only difference is that it can only recover 3 items (300%).

Metal Scrap - created by Journeyman crafting, using one iron ingot, you will have to fill the resulting base with any three iron objects, after which you will receive a ready repair kit. He can recover 4 items (400%). Requires a table for repair.

This will make blacksmiths a slightly more useful role, as well as teach players to stock up on consumables in order to be less dependent on virtue (blacksmith/squire) when playing classes wearing metal armor.

Squires get good repair kits the roundstart. The Abandoned Squire's Virtue also adds 2 good repair kits to the storage.

Ironclad adventurer get one low metal-kit.

Ready up bonus give you one low metal kit (if you has trait medium/plate training), one low fabric kit.

Add repair kits on dungeon loot spawners.
Add new, additional craft for low-quality metal kit. (use 3 scrap instead 1 iron bar)
Add new item, "scrap". (you can take it from fully destroyed iron armor or weapons, or just click on it with stake in your hand)
Add low-quality repair kits ability to spawn on medium and heavy skeletons. (15%)

<img width="193" height="195" alt="image" src="https://github.com/user-attachments/assets/47d4f09b-229f-45c7-aae9-9919f0b81fa9" />

## Testing Evidence

- [x] Tested on local game.

## Why It's Good For The Game

No matter how much you want to protect the mythical interaction between players, a character clad in metal armor is very dependent on the possibility of repair. So much so that many players prefer to take on the blacksmith's or squire's virtue, just to be able to repair themselves after a skirmish with skeletons, rather than walking around with half-broken armor. The PR will not diminish the influence of blacksmiths. It adds 2 options. You can create poor repair kits yourself, in the field, to quickly patch up your armor, or buy high-quality ones from the city's specialists. Let's be honest, a soldier who can't maintain his equipment himself, at least on a primitive level, is shit, not a soldier.

Dark-Umbrella committed:
repair kits

Dark-Umbrella committed:
Update repair_kits.dm

Dark-Umbrella committed:
squire

Dark-Umbrella committed:
bandits

Dark-Umbrella commented:
Only sewing and forging of various kinds. Carpentry is a fairly rare skill for repairs, and it can be learned many times faster by simply crafting chests, no matter how terrible it may sound. I haven't thought about adding such sets, but maybe I'll make another type specifically for carpentry if there are more items using it in the game, not just two types of shields, staffs, and elf mercenary armor.

Ollanius commented:
It's all over the monitor, sire.
I love this a lot - both in terms of presentation and proposals.

Aisu9 commented:
wait you gate repair from broken to crafter only? this is a terrible idea are you aware how fast equipement break while exploring?
if you aim to continue on this way please separate repair skill from crafting skill and make it uncapped

radarseas commented:
there's also l the point of mending spell, it is **required** to be able to fix fully broken items else Avantyne/guilded armor users can't fix their own armor.

But it will also outclass self-repair a billionfold and mages are already extremely more viable than most adventuring classes due to their hybrid ranged/melee abilities + crowd control - especally if you need expert skill to fully repair stuff.

---

Despite this though, I absolutely adore the concept here. A lot. - Perhaps have it so you can FULLY repair something back to unbroken again with low armorsmithing/weaponsmithing skills but only like 20% at most unless you're using a kit. So people can still keep their gear intact in the field.

Dakken243 commented:
> Only sewing and forging of various kinds. Carpentry is a fairly rare skill for repairs, and it can be learned many times faster by simply crafting chests, no matter how terrible it may sound. I haven't thought about adding such sets, but maybe I'll make another type specifically for carpentry if there are more items using it in the game, not just two types of shields, staffs, and elf mercenary armor.

Please do either make a repair kit or remove the expert requirement from carpentry items, is all I can say. Fucking over an entire section of ppl just bc the list isnt big enough to 'warrant it' ain't great.

Especially when, as you said yourself. It includes class/role unique items like the armor.

Aisu9 commented:
there is a "fine" line between making 1 virtue MANDATORY and making said virtue just a **nice** plus

HeroWithYay commented:
Please for the love of god can this not be merged until the ridiculous skill requirements for hammer repairs are removed. Repairing is tedious enough.

Repair kits could be a nice way to SITUATIONALLY alleviate the tedium of having to grind repair skills each week or burn a virtue, for characters that can afford it.
But this just makes it so, so much worse.

Gear fully breaks so often. Like literally every encounter with a non-simplemob. Do not make the existing repair worse.

YakumoChen commented:
> Also: Full-plate is create-able at journeyman. Why is a good repair kit requiring expert? You cannot be telling me it is harder to make a repair kit than it is to forge a full set of fullplate.

Maybe the best armours in the game should be locked to the Smith or silverface?

The better repair kits skill requirement seems tailored to give towner roles something exclusive to fill the peddler with, which is fine. They can make the field repair kits instead, but I agree the nerf to skill is a little harsh, it could maybe be toned down or the repair kits could be made more appealing somehow. Maybe field repairs could cap out without a kit or higher skill levels (or the squire perk) instead of being shunted to the shadow realm of expert

Dark-Umbrella committed:
A plastical world wins...

HeroWithYay commented:
> > Also: Full-plate is create-able at journeyman. Why is a good repair kit requiring expert? You cannot be telling me it is harder to make a repair kit than it is to forge a full set of fullplate.
> 
> Maybe the best armours in the game should be locked to the Smith or silverface?
> 
> The better repair kits skill requirement seems tailored to give towner roles something exclusive to fill the peddler with, which is fine. They can make the field repair kits instead, but I agree the nerf to skill is a little harsh, it could maybe be toned down or the repair kits could be made more appealing somehow. Maybe field repairs could cap out without a kit or higher skill levels (or the squire perk) instead of being shunted to the shadow realm of expert

I think repairing just doesn't need to be made harder fullstop. There's plenty of use for these repair kits as an alternate way to repair for characters that dont want to take the virtue. But why screw over anyone who is taking the virtue/playing squire errant?

Like I don't think a singular new consumeable needs to warp the entire game to revolve around it.

Dark-Umbrella committed:
silverface

Dark-Umbrella committed:
Update needle.dm

Dark-Umbrella committed:
fix literacy, add low-quality repair kits to ready up bonus and add one for ironclad

Dark-Umbrella committed:
Check Description

Add repair kits on dungeon loot spawners.
Add new, additional craft for low-quality metal kit. (use 3 scrap instead 1 iron bar)
Add new item, "scrap". (you can take it from fully destroyed iron armor or weapons, or just click on it with stake in your hand)
Add low-quality repair kits ability to spawn on medium and heavy skeletons. (15%)
